src/Raven.Server/Documents/Transformers/TransformerStore.cs

```diff
@@ -0,0 +1,181 @@
+﻿using System;
+using System.Diagnostics;
+using System.IO;
+using System.Threading.Tasks;
+using Raven.Abstractions.Data;
+using Raven.Abstractions.Indexing;
+using Raven.Abstractions.Logging;
+using Sparrow;
+
+using Voron.Platform.Posix;
+
+namespace Raven.Server.Documents.Transformers
+{
+    public class TransformerStore : IDisposable
+    {
+        private static readonly ILog Log = LogManager.GetLogger(typeof(TransformerStore));
+
+        private readonly DocumentDatabase _documentDatabase;
+
+        private readonly CollectionOfTransformers _transformers = new CollectionOfTransformers();
+
+        private readonly object _locker = new object();
+
+        private bool _initialized;
+
+        private string _path;
+
+        public TransformerStore(DocumentDatabase documentDatabase)
+        {
+            _documentDatabase = documentDatabase;
+        }
+
+        public Task InitializeAsync()
+        {
+            if (_initialized)
+                throw new InvalidOperationException($"{nameof(TransformerStore)} was already initialized.");
+
+            lock (_locker)
+            {
+                if (_initialized)
+                    throw new InvalidOperationException($"{nameof(TransformerStore)} was already initialized.");
+
+                if (_documentDatabase.Configuration.Indexing.RunInMemory == false)
+                {
+                    _path = Path.Combine(_documentDatabase.Configuration.Indexing.IndexStoragePath, "Transformers");
+
+                    if (Platform.RunningOnPosix)
+                        _path = PosixHelper.FixLinuxPath(_path);
+
+                    if (Directory.Exists(_path) == false && _documentDatabase.Configuration.Indexing.RunInMemory == false)
+                        Directory.CreateDirectory(_path);
+                }
+
+                _initialized = true;
+
+                return Task.Factory.StartNew(OpenTransformers, TaskCreationOptions.LongRunning);
+            }
+        }
+
+        private void OpenTransformers()
+        {
+            if (_documentDatabase.Configuration.Indexing.RunInMemory)
+                return;
+
+            lock (_locker)
+            {
+                foreach (var transformerFile in new DirectoryInfo(_path).GetFiles("*.transformer"))
+                {
+                    if (_documentDatabase.DatabaseShutdown.IsCancellationRequested)
+                        return;
+
+                    int transformerId;
+                    if (int.TryParse(transformerFile.Name, out transformerId) == false)
+                        continue;
+
+                    var transformer = Transformer.Open(transformerId, _documentDatabase.Configuration.Indexing);
```

Comment:
If a single transformer throws, we want to mark it as invalid, and not kill the startup of the whole server